src/memo/memo_intf.ml

```diff
@@ -1,5 +1,10 @@
 open Stdune
 
+module type Sexpable = sig
```

Comment:
A bit of a nit, but I think we should just move this to `Stdune.Sexp_intf`.

Reply:
I don't know. `Sexp_intf` doesn't look like a module for collecting module types like this (it currently doesn't have any), and I don't enjoy this indirection. The only reason I'm giving it a name at all is that OCaml won't let you have anonymous first-class module types.

In fact I just pushed a commit that removes "memo_intf" altogether, moving these definitions into `memo.ml{,i}` to make things less confusing.

Reply:
Well, it's just a general purpose module type that doesn't have much to do with memoization. Usually, I prefer to keep such code in the stdlib, or other purpose general purpose libraries. Obviously, there's some personal discretion involved, but usually it's a good thing.

I'm not sure `Sexp_intf` is the right place for this module type either, so if you have a better spot feel free to suggest it.

Reply:
The signature - yes, the name - not clear. In fact this name means something else in `Core` so I hesitate to increase its scope.
I asked @diml  and he says "why don't we call things for what they are used for instead". I'm not a big fan of that either because I like things named for what they are rather than why they exist, but it seems OK to me.

I renamed the module types to:

````
Output_simple
Output_allow_cutoff
Input
````

and I removed the hash from "allow_cutoff".